### PR TITLE
ARROW-11418: [Doc] Add buffer compression to IPC support matrix

### PR DESCRIPTION
**Pull request by pitrou**
Also fix a couple warnings and reword description of PyArrow serialization.

### CODE REVIEW

File: docs/source/status.rst [hunk 1]
```diff
@@ -126,6 +126,8 @@ IPC Format
 +-----------------------------+-------+-------+-------+------------+-------+-------+-------+
 | Sparse tensors              | ✓     |       |       |            |       |       |       |
 +-----------------------------+-------+-------+-------+------------+-------+-------+-------+
+| Buffer compression          | ✓     |       |       |            |       |       | ✓     |
++-----------------------------+-------+-------+-------+------------+-------+-------+-------+
```
Comment (pitrou) [inline docs/source/status.rst:130]: @quinnj AFAICT, Julia supports IPC buffer compression?
@nevi-me AFAICT, Rust doens't support IPC buffer compression?

Comment (nevi-me) [reply docs/source/status.rst:130]: No Rust support yet. Came across issues while working on it, but now that there are test files, I'll be able to finish the support

Comment (quinnj) [reply docs/source/status.rst:130]: Yes, Julia supports IPC buffer compression/decompression for lz4_frame and zstd.

**Comment by pitrou** [comment]
+1, will merge.

**Pull request closed without merging**